### PR TITLE
deployment/ccip/changeset: add RevokeCandidate cs

### PR DESCRIPTION
**Pull request by makramkd**
Add the revoke candidate changeset for CCIPHome w/ optional MCMS, along
with a basic unit test.

### CODE REVIEW

File: deployment/ccip/changeset/cs_ccip_home.go [hunk 1]
```diff
@@ -647,3 +647,189 @@ func promoteAllCandidatesForChainOps(
...
+	NodeIDs          []string
+	PluginType       types.PluginType
+
+	// MCMS is optional MCMS configuration, if provided the changeset will generate an MCMS proposal.
+	// If nil, the changeset will execute the commands directly using the deployer key
+	// of the provided environment.
+	MCMS *MCMSConfig
+}
+
+func (r RevokeCandidateChangesetConfig) Validate(e deployment.Environment, state CCIPOnChainState) (deployment.Nodes, error) {
```
Comment (AnieeG) [inline deployment/ccip/changeset/cs_ccip_home.go]: if we at all put the `validate` as part of interface (to ensure `Validate` is called on all configs), the standardization of method signature would be difficult. Can we get nodes as part of a diff method?

Comment (makramkd) [reply deployment/ccip/changeset/cs_ccip_home.go]: I think it'd be tough to have it as part of the interface tbh, the family of validation funcs is already all over the place.

File: deployment/ccip/changeset/cs_ccip_home.go [hunk 2]
```diff
@@ -272,6 +269,9 @@ func (s SetCandidateChangesetConfig) Validate(e deployment.Environment, state CC
 		return nil, fmt.Errorf("invalid ccip ocr params: %w", err)
 	}
 
+	// TODO: validate token config in the commit config, if commit is the plugin.
+	// TODO: validate gas config in the chain config in cciphome for this DONChainSelector.
```
Comment (makramkd) [inline deployment/ccip/changeset/cs_ccip_home.go:246]: Created tickets for these: 
* https://smartcontract-it.atlassian.net/browse/CCIP-4615 token config validation
* https://smartcontract-it.atlassian.net/browse/CCIP-4616 gas config validation

File: deployment/ccip/changeset/cs_ccip_home.go [hunk 3]
```diff
@@ -174,36 +173,10 @@ func PromoteAllCandidatesChangeset(
 	}, nil
 }
 
-// AddDonAndSetCandidateChangesetConfig is a separate config struct
```
Comment (makramkd) [inline deployment/ccip/changeset/cs_ccip_home.go:177]: This was just moved further down closer to the changeset that uses it

File: deployment/ccip/changeset/cs_ccip_home_test.go [hunk 4]
```diff
@@ -4,272 +4,20 @@ import (
...
 
-	"github.com/smartcontractkit/chainlink/deployment"
-
 	"github.com/stretchr/testify/require"
 
 	commonchangeset "github.com/smartcontractkit/chainlink/deployment/common/changeset"
 	"github.com/smartcontractkit/chainlink/deployment/common/proposalutils"
 )
 
-func TestActiveCandidate(t *testing.T) {
```
Comment (makramkd) [inline deployment/ccip/changeset/cs_ccip_home_test.go:29]: Removed this test for now as maintaining it is an unnecessary burden, we need to rewrite it, ticket here: https://smartcontract-it.atlassian.net/browse/CCIP-3995

**Review by connorwstein — APPROVED** [review]

**Review by AnieeG — APPROVED** [review]

**Pull request merged**